Cabal/Distribution/ModuleName.hs

```diff
@@ -76,12 +76,12 @@ simple str = ModuleName (stlFromStrings [str])
 -- an error if it is used with a string that is not a valid module name. If you
 -- are parsing user input then use 'Distribution.Text.simpleParse' instead.
 --
-fromString :: String -> ModuleName
-fromString string = fromComponents (split string)
-  where
-    split cs = case break (=='.') cs of
-      (chunk,[])     -> chunk : []
-      (chunk,_:rest) -> chunk : split rest
+instance IsString ModuleName where
+    fromString string = fromComponents (split string)
+      where
+        split cs = case break (=='.') cs of
```

Comment:
Now i remember, `splitOn` is quite general reusing other combinators in `split`. Also this is less general `splitOn' :: a -> [a] -> [[a]]`, where general one splits on `[a]`.